src/codeflare_sdk/ray/rayjobs/runtime_env.py

```diff
@@ -0,0 +1,408 @@
+    elif isinstance(pip_spec, str):
+        # Assume it's a path to requirements.txt
+        return parse_requirements_file(pip_spec)
+    elif isinstance(pip_spec, dict):
+        # Handle dict format (e.g., {"packages": [...], "pip_check": False})
+        if "packages" in pip_spec:
+            logger.info(
+                f"Using pip dependencies from dict: {len(pip_spec['packages'])} packages"
+            )
+            return pip_spec["packages"]
+
+    logger.warning(f"Unsupported pip specification format: {type(pip_spec)}")
+    return None
+
+
+def parse_requirements_file(requirements_path: str) -> Optional[List[str]]:
+    """
+    Parse a requirements.txt file and return list of dependencies.
+
+    Args:
+        requirements_path: Path to requirements.txt file
+
+    Returns:
+        List of pip dependencies
+    """
+    if not os.path.isfile(requirements_path):
+        logger.warning(f"Requirements file not found: {requirements_path}")
+        return None
+
+    try:
+        with open(requirements_path, "r") as f:
+            lines = f.readlines()
+
+        # Parse requirements, filtering out comments and empty lines
+        requirements = []
+        for line in lines:
+            line = line.strip()
+            if line and not line.startswith("#"):
+                requirements.append(line)
+
+        logger.info(f"Parsed {len(requirements)} dependencies from {requirements_path}")
+        return requirements
+
+    except (IOError, OSError) as e:
+        logger.warning(f"Could not read requirements file {requirements_path}: {e}")
+        return None
+
+
+def create_secret_from_spec(
+    job: RayJob, secret_spec: Dict[str, Any], rayjob_result: Dict[str, Any] = None
+) -> str:
+    """
+    Create Secret from specification via Kubernetes API.
+
+    Args:
+        secret_spec: Secret specification dictionary
+        rayjob_result: The result from RayJob creation containing UID
+
+    Returns:
+        str: Name of the created Secret
+    """
+
+    secret_name = secret_spec["metadata"]["name"]
+
+    metadata = client.V1ObjectMeta(**secret_spec["metadata"])
+
+    # Add owner reference if we have the RayJob result
+    if (
+        rayjob_result
+        and isinstance(rayjob_result, dict)
+        and rayjob_result.get("metadata", {}).get("uid")
+    ):
+        logger.info(
+            f"Adding owner reference to Secret '{secret_name}' with RayJob UID: {rayjob_result['metadata']['uid']}"
+        )
+        metadata.owner_references = [
+            client.V1OwnerReference(
+                api_version="ray.io/v1",
+                kind="RayJob",
+                name=job.name,
+                uid=rayjob_result["metadata"]["uid"],
+                controller=True,
+                block_owner_deletion=True,
+            )
+        ]
+    else:
+        logger.warning(
+            f"No valid RayJob result with UID found, Secret '{secret_name}' will not have owner reference. Result: {rayjob_result}"
```

Comment:
How often do we expect this to happen? Better to have the owner reference so that the secret is removed if the owning resource is removed.